### PR TITLE
Add surface and toa flux variables to fine res budget dataset

### PR DESCRIPTION
**Pull request by mcgibbon**
This PR adds surface fluxes of precip, latent and sensible heat, and radiation, and top of atmosphere fluxes of radiation, to the fine_res_budget dataset.

Refactored public API:
- added surface fluxes of precip, latent and sensible heat, and radiation, and top of atmosphere fluxes of radiation, to the fine_res_budget dataset
- 2D gfs physics variables can now be added to the fine_res_budget workflow as dataset outputs

Resolves #<github issues> [JIRA-TAG]

(Delete this for the commit message)
You are encouraged to check the PR against the [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--A4lKrs~xg7w5Gsb39N6JLNQoAg-IlsYffZgTwyKEylty7NhY).

### CODE REVIEW

File: external/loaders/loaders/mappers/_fine_res_budget.py [hunk 1]
```diff
@@ -44,6 +44,16 @@ class FineResBudget(Protocol):
...
+    DLWRFsfc_coarse: xarray.DataArray
+    DSWRFsfc_coarse: xarray.DataArray
+    DSWRFtoa_coarse: xarray.DataArray
+    ULWRFsfc_coarse: xarray.DataArray
+    ULWRFtoa_coarse: xarray.DataArray
+    USWRFsfc_coarse: xarray.DataArray
+    USWRFtoa_coarse: xarray.DataArray
+    LHTFLsfc_coarse: xarray.DataArray
+    SHTFLsfc_coarse: xarray.DataArray
+    PRATEsrf_coarse: xarray.DataArray
```
Comment (oliverwm1) [inline external/loaders/loaders/mappers/_fine_res_budget.py:56]:
```suggestion
    PRATEsfc_coarse: xarray.DataArray
```

File: workflows/fine_res_budget/budget/budgets.py [hunk 2]
```diff
@@ -50,8 +50,10 @@ def pressure_level_average(self, delp_fine, delp_coarse, area, field, factor):
         pi = self.pressure_at_interface(delp_fine)
         pi_c = self.pressure_at_interface(delp_coarse)
         pi_c_up = self.block_upsample(pi_c, factor=factor)
-
-        fg = self.regrid_vertical(pi, field, pi_c_up)
+        if self.z in field.dims:
+            fg = self.regrid_vertical(pi, field, pi_c_up)
+        else:
+            fg = field
```
Comment (oliverwm1) [inline workflows/fine_res_budget/budget/budgets.py:56]: Doing a "pressure level average" of a 2D field doesn't make sense. What about moving this `if` statement one level up and defining a 2D coarsening function that does not take in the pressure thicknesses or call `regrid_vertical`?

Or at the very least, include lines 50-52 inside the `if self.z in field.dims:` block

Comment (mcgibbon) [reply workflows/fine_res_budget/budget/budgets.py:56]: Updated. Moved this up one level, makes a lot more sense.

**Review by oliverwm1 — APPROVED** [review]
LGTM

**Pull request merged**